spec/blocks-test.js

```diff
@@ -370,6 +377,67 @@ describe('The CodeMirrorBlocks Class', function() {
           // TODO: figure out how to test this.
         });
       });
+
+      describe('tree navigation', function() {
+        beforeEach(function() {
+          this.cm.setValue('(+ 1 2 3) 99');
+          this.firstRoot = this.blocks.ast.rootNodes[0];
+          this.secondRoot = this.blocks.ast.rootNodes[1];
+          this.funcSymbol = this.blocks.ast.rootNodes[0].func;
+          this.firstArg = this.blocks.ast.rootNodes[0].args[0];
+          this.secondArg = this.blocks.ast.rootNodes[0].args[1];
+          this.thirdArg = this.blocks.ast.rootNodes[0].args[2];
+        });
+
+        it('left-arrow should navigate to the next sibling, but not beyond it', function() {
+          this.secondRoot.el.dispatchEvent(click());
+          expect(document.activeElement).toBe(this.secondRoot.el);
+          expect(this.blocks.scroller.getAttribute('aria-activedescendent')).toBe(this.secondRoot.el.id);
+          this.firstRoot.el.dispatchEvent(keydown(37));
```

Comment:
you could put these key codes in a const to make it easier to understand what is going on.